packages/@o3r/core/schematics/ng-add-create/index.ts

```diff
@@ -0,0 +1,74 @@
+import { chain, Rule, SchematicContext, Tree } from '@angular-devkit/schematics';
+import { getTemplateFolder } from '@o3r/schematics';
+import * as path from 'node:path';
+import * as fs from 'node:fs';
+import type {PackageJson} from 'type-fest';
+
+import { apply, MergeStrategy, mergeWith, renameTemplateFiles, template, url } from '@angular-devkit/schematics';
+import { NgGenerateUpdateSchematicsSchema } from './schema';
+
+/**
+ * Rule factory to include `ng add` skelethon
+ *
+ * @param options
+ * @param rootPath
+ */
+export function updateTemplates(options: NgGenerateUpdateSchematicsSchema, rootPath: string): Rule {
+
+  return (tree: Tree, context: SchematicContext) => {
+
+    // register scripts
+    if (tree.exists('/package.json')) {
+      const packageJson: PackageJson = JSON.parse(tree.read('/package.json')!.toString());
+      if (!packageJson.devDependencies?.['@angular/cli']) {
+        context.logger.info('You are not in an angular project. Ng add skelethon creation aborted.');
+        return tree;
+      }
+      const o3rCorePackageJsonPath = path.resolve(__dirname, '..', '..', 'package.json');
+      const o3rCorePackageJson: PackageJson = JSON.parse(fs.readFileSync(o3rCorePackageJsonPath)!.toString());
+      // prepare needed deps for schematics
+      const angularVersion = packageJson.devDependencies['@angular/cli'];
+      const otterVersion = o3rCorePackageJson.dependencies!['@o3r/schematics'];
+
+      packageJson.scripts ||= {};
+      packageJson.scripts['prepare:build:builders'] = 'yarn cpy \'schematics/**/*.json\' dist/schematics && yarn cpy \'collection.json\' dist';
+      packageJson.scripts['build:builders'] = 'tsc -b tsconfig.builders.json --pretty';
```

Comment:
these scripts are not used out of the box, an additional manual step is needed to update the build process / pipepline to include them in the published artifact.
it might be tricky to support this but we could just add some doc to explain how to finalize the setup maybe